### DISCUSSION TITLE
Is there a better choice than `sleep +Inf` for a "null" background container run cmd?

### DISCUSSION THREAD

**Discussion by castedo** [containers/podman#12934] [Q&A]
On my desktop, I'm usually executing multiple processes in the same container via `podman exec` for a variety of reasons (mainly because I use https://cnest.readthedocs.io). More often than not, the initial process started via `podman start` (or `podman run`) is to do absolutely nothing other than run in the background and wait to be stopped.

To date, I've been running `sleep +Inf` as this "null" background process. But I get a side effect of needing to run with `--pid=host`. If I don't then `podman stop` takes 10 seconds, whereas with `--pid=host` it stops immediately. Detailed example below.

**Is there a better and more proper choice for background process?** I actually don't mind using `--pid=host` but somehow `sleep +Inf` smells like there's some better approach I should be taking for a "null" background process.

(Note: I understand that Linux namespaces need some process. Regardless, I'm often finding myself having no use for this process other than to just have a process for the Linux namespace.)

```
[castedo@nasa ~]$ podman run -d --name so-slow centos sleep +Inf
b37107cf81d60c491218520be9f90ecabeddc282e79028d8a91388894688f353
[castedo@nasa ~]$ time podman stop so-slow
so-slow

real    0m10.317s
user    0m0.072s
sys 0m0.051s
[castedo@nasa ~]$ podman run -d --pid=host --name so-fast centos sleep +Inf
01cee649d314e32486696ad99c1f18f5d57b6fb8f95f949afa73cdfff5464d81
[castedo@nasa ~]$ time podman stop so-fast
WARN[0000] cannot toggle freezer: cgroups not configured for container
WARN[0000] cannot toggle freezer: cgroups not configured for container
WARN[0000] lstat : no such file or directory            
so-fast

real    0m0.354s
user    0m0.057s
sys 0m0.043s
[castedo@nasa ~]$  
```

```
[castedo@nasa ~]$ head -2 /etc/os-release
NAME="Red Hat Enterprise Linux"
VERSION="8.5 (Ootpa)"
[castedo@nasa ~]$ rpm -q podman
podman-3.3.1-9.module+el8.5.0+12697+018f24d7.x86_64
```

**Comment by zhangguanzhang** [comment]
https://stackoverflow.com/questions/2935183/bash-infinite-sleep-infinite-blocking

**Comment by Luap99** [comment]
You can run `podman stop -t 0` or  you could also create the container with `--stop-timeout 0` or `--stop-signal SIGKILL`. sleep will ignore SIGTERM and the default timeout is 10 seconds before podman uses SIGKILL.
